server/src/internalClusterTest/java/org/elasticsearch/snapshots/ConcurrentSnapshotsIT.java

```diff
@@ -1112,41 +1112,52 @@ public void testConcurrentOperationsLimit() throws Exception {
 
         blockNodeOnAnyFiles(repoName, masterName);
         int blockedSnapshots = 0;
-        boolean blockedDelete = false;
         final List<ActionFuture<CreateSnapshotResponse>> snapshotFutures = new ArrayList<>();
         ActionFuture<AcknowledgedResponse> deleteFuture = null;
         for (int i = 0; i < limitToTest; ++i) {
-            if (blockedDelete || randomBoolean()) {
+            if (deleteFuture != null || randomBoolean()) {
                 snapshotFutures.add(startFullSnapshot(repoName, "snap-" + i));
                 ++blockedSnapshots;
             } else {
-                blockedDelete = true;
                 deleteFuture = startDeleteSnapshot(repoName, randomFrom(snapshotNames));
             }
         }
         awaitNumberOfSnapshotsInProgress(blockedSnapshots);
-        if (blockedDelete) {
+        if (deleteFuture != null) {
             awaitNDeletionsInProgress(1);
         }
         waitForBlock(masterName, repoName);
 
-        final String expectedFailureMessage = "Cannot start another operation, already running [" + limitToTest +
-            "] operations and the current limit for concurrent snapshot operations is set to [" + limitToTest + "]";
-        final ConcurrentSnapshotExecutionException csen1 = expectThrows(ConcurrentSnapshotExecutionException.class,
+        final ConcurrentSnapshotExecutionException cse = expectThrows(ConcurrentSnapshotExecutionException.class,
             () -> client().admin().cluster().prepareCreateSnapshot(repoName, "expected-to-fail").execute().actionGet());
-        assertThat(csen1.getMessage(), containsString(expectedFailureMessage));
-        if (blockedDelete == false || limitToTest == 1) {
-            final ConcurrentSnapshotExecutionException csen2 = expectThrows(ConcurrentSnapshotExecutionException.class,
-                () -> client().admin().cluster().prepareDeleteSnapshot(repoName, "*").execute().actionGet());
-            assertThat(csen2.getMessage(), containsString(expectedFailureMessage));
+        assertThat(cse.getMessage(), containsString("Cannot start another operation, already running [" + limitToTest +
+                "] operations and the current limit for concurrent snapshot operations is set to [" + limitToTest + "]"));
+        boolean deleteAndAbortAll = false;
+        if (deleteFuture == null && randomBoolean()) {
```

Comment:
I wonder if the case of having a specific delete and a "*" delete concurrently is not worth testing?

Reply:
We do test this and other such snapshot delete batching/concurrency things in this class but I think with the removal of the limit check for deletes altogether it's not really adding any coverage here.